### PR TITLE
[azure] refacto

### PR DESCRIPTION
**Pull request by vysahjk**

### CODE REVIEW

File: Babylon/commands/azure/ad/app/create.py [hunk 1]
```diff
@@ -1,45 +1,84 @@
...
+    if "id" in output_data:
+        return True
+
+
+def is_correct_response_serviceprincipal(response):
+    if response is None:
+        return CommandResponse.fail()
+    output_data = response.json()
+    if "id" in output_data:
+        return True
```
Comment (enocquet) [inline Babylon/commands/azure/ad/app/create.py:84]: I think here we could go for a unique function, let's call it 'is_correct_response' which responsibility would be to return a boolean if the given condition is met

File: Babylon/commands/azure/adt/instance/create.py [hunk 2]
```diff
@@ -53,56 +55,73 @@ def create(
...
+                    "/subscriptions/" + azure_subscription + "/providers/Microsoft.Authorization/roleDefinitions/" +
+                    adt_data_owner_role_id,
+                "principalId":
+                    context['platform_principal_id'],
+                "principalType":
+                    "ServicePrincipal",
+            },
+        )
+    except Exception as e:
+        logger.error(f"Failed to assign a new role to ADT instance '{name}': {e}")
```
Comment (enocquet) [inline Babylon/commands/azure/adt/instance/create.py:119]: Maybe we could save some lines of code like this:
```python
    for principal_id in ['babylon_principal_id', 'platform_principal_id']:
        try:
            authorization_client.role_assignments.create(
                scope,
                str(uuid.uuid4()),
                {
                    "roleDefinitionId":
                        "/subscriptions/" + azure_subscription + "/providers/Microsoft.Authorization/roleDefinitions/" +
                        adt_data_owner_role_id,
                    "principalId":
                        context[principal_id],
                    "principalType":
                        "ServicePrincipal",
                },
            )
        except Exception as e:
            logger.error(f"Failed to assign a new role to ADT instance '{name}': {e}")
```

File: Babylon/commands/azure/adx/consumer/add.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Create new consumer group in EventHub
+    """
+    check_ascii(name)
+    azure_subscription = context['azure_subscription_id']
+    rg = context['azure_resource_group_name']
+    org_id = context['api_organization_id'].lower()
+    work_id = context['api_workspace_key'].lower()
+    location = context['azure_resource_location']
+
+    client = EventHubManagementClient(credential=DefaultAzureCredential(), subscription_id=azure_subscription)
```
Comment (enocquet) [inline Babylon/commands/azure/adx/consumer/add.py:38]: What is the expected behavior here ? Do we need to have the classical AZURE_CLIENT_ID, AZURE_CLIENT_SECRET and AZURE_SUBSCRIPTION_ID to be set as environment variables ?

**Review by enocquet — CHANGES_REQUESTED** [review]

File: Babylon/commands/azure/adt/instance/create.py (hunk 2 above)
Comment (vysahjk) [reply Babylon/commands/azure/adt/instance/create.py:119]: yes, thanks. done!

File: Babylon/commands/azure/adx/consumer/add.py (hunk 3 above)
Comment (vysahjk) [reply Babylon/commands/azure/adx/consumer/add.py:38]: yep, i think is a 🐞 ... 

maybe:

`from Babylon.utils.clients import get_azure_credentials`
`client = EventHubManagementClient(credential=get_azure_credentials())` ?

Comment (vysahjk) [reply Babylon/commands/azure/adx/consumer/add.py:38]: after test: 

**subscription_id** is required in EventHubManagementClient class

**Review by enocquet — CHANGES_REQUESTED** [review]

File: Babylon/commands/azure/func/deploy.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+env = Environment()
+
+
+@command()
+@pass_arm_client
+@option("--complete-mode", "deploy_mode_complete", is_flag=True)
+@option("-f", "--file", "deploy_file", type=Path(readable=True, dir_okay=False, path_type=pathlib.Path))
+@argument("deployment_name", type=QueryType())
+@inject_context_with_resource({'api': ['organization_id', 'workspace_key'], 'azure': ['resource_group_name']})
+def deploy(
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/azure/func/deploy.py:30]: yes, specific to webapp deployment (Azure Function)

File: Babylon/commands/azure/arm/run.py [hunk 5]
```diff
@@ -1,56 +1,86 @@
...
+        # deployments created
+
     except HttpResponseError as _e:
         logger.error(f"An error occurred : {_e.message}")
         return CommandResponse.fail()
 
-    logger.info("Deployment created")
-
+    _ret: list[str] = ["Provisioning state: successful"]
+    logger.info("\n".join(_ret))
```
[the remark this answers is not in the document]
Comment (vysahjk) [reply Babylon/commands/azure/arm/run.py]: done

**Review by enocquet — APPROVED** [review]

**Pull request merged**